cogs/tictactoecog.py

```diff
@@ -0,0 +1,205 @@
+from discord.ext import commands
+
+class TicTacToeCog(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command()
+    async def tictactoe(self, ctx):
+        """Plays a game of tic tac toe with the user (it's unbeatable)"""
+
+        #Initialize list to hold the board numbers/symbols
+        board = []
+
+        #Displays the board nicely using f-strings and accessing the indices
+        async def display_board():
+            await ctx.send(f'```py\n {board[0]} | {board[1]} | {board[2]}\n'
+            + f'-----------\n {board[3]} | {board[4]} | {board[5]}\n'
+            + f'-----------\n {board[6]} | {board[7]} | {board[8]}```')
+        
+        
+        #Checks if the inputted player has won
+        def check_win(player):
+            #These are all the winning lines/combinations
+            lines = [(0, 1, 2), (3, 4, 5), (6, 7, 8), 
+                    (0, 3, 6), (1, 4, 7), (2, 5, 8),
+                    (0, 4, 8), (2, 4, 6)]
+            
+            #For each one, if all 3 of those positions are the player's character, he's won
+            for line in lines:
+                if board[line[0]] == board[line[1]] == board[line[2]] == player:
+                    return True
+
+            #If I get to the end and he hasn't won, I can return false
+            return False
+        
+        
+        #Simply makes a move given a player and their move
+        def make_move(player, space):
+            
+            #Convert space to int, and decrement for 0 based indexing, set to player
+            board[int(space)-1] = player
+
+
+        #Resets a move (will use for recursive backtracking)
+        def remove_move(space):
+
+            #Set that board space back to its number
+            board[int(space)-1] = space
+
+        
+        #Returns a list of all valid moves in the position
+        def valid_moves():
+            moves = []
+
+            #Go through each space, if it's not X or O, append it
+            for space in board:
+                if space != 'X' and space != 'O':
+                    moves.append(space)
+            
+            #Return moves at the end
+            return moves
+
+
+        #Checks if there's a draw on the board
+        def check_draw():
+
+            #If no valid moves, true, otherwise false
+            if len(valid_moves()) == 0:
```

Comment:
You don't need an if statement here, you can just `return len(valid_moves()) == 0`